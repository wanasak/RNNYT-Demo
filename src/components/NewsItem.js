import React, { Component } from "react";
import PropTypes from "prop-types";
import {
    View,
    StyleSheet,
    TouchableOpacity,
    ActionSheetIOS
} from "react-native";
import AppText from "./AppText";
import Thumbnail from "./Thumbnail";
import ByLine from "./ByLine";
import * as globalStyles from "../styles/global";

export default class NewsItem extends Component {
    constructor(props) {
        super(props);

        this.onLongPress = this.onLongPress.bind(this);
    }

    onLongPress() {
        ActionSheetIOS.showActionSheetWithOptions(
            {
                options: ["Bookmark", "Cancel"],
                cancelButtonIndex: 1,
                title: this.props.title
            },
            buttonIndex => {
                if (buttonIndex === 0) {
                    this.props.onBookmark();
                }
            }
        );
    }

    render() {
        const {
            style,
            imageUrl,
            title,
            author,
            date,
            location,
            description,
            onPress
        } = this.props;
        const accentColor =
            globalStyles.ACCENT_COLORS[
                this.props.index % globalStyles.ACCENT_COLORS.length
            ];

        return (
            <TouchableOpacity
                onPress={onPress}
                onLongPress={this.onLongPress}
                style={style}
            >
                <View>
                    <Thumbnail
                        url={imageUrl}
                        titleText={title}
                        accentColor={accentColor}
                        style={styles.thumbnail}
                    />
                    <View style={styles.content}>
                        <ByLine
                            author={author}
                            date={date}
                            location={location}
                        />
                        <AppText>{description}</AppText>
                    </View>
                </View>
            </TouchableOpacity>
        );
    }
}

NewsItem.propType = {
    imageUrl: PropTypes.string,
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    date: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    location: PropTypes.string,
    index: PropTypes.number.isRequired,
    onPress: PropTypes.func.isRequired,
    style: View.propTypes.style,
    onPress: PropTypes.func.isRequired,
    onBookmark: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
    thumbnail: {
        marginBottom: 5
    },
    content: {
        paddingHorizontal: 5
    }
});
