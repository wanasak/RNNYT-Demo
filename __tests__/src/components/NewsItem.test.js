import "react-native";
import React from "react";
import NewsItem from "../../../src/components/NewsItem";
import renderer from 'react-test-renderer';

// a function that does nothing
const noop = () => {};

const testData = {
    description:
        "React Native, the framework for building mobile applications with web technologies, is expanding to new platforms.",
    author: "JACOB FRIEDMANN",
    location: "",
    imageUrl: "https://example.com/image.jpg",
    date: "Sep 10th 2016",
    title: "React Native Expands to New Platforms",
    url: "http://example.com/react-native-expands",
    onPress: noop,
    onBookmark: noop
};

describe("NewsItem component", () => {
    it("should render correctly", () => {
        const renderedComponent = renderer.create(
            <NewsItem
                index={0}
                {...testData}
            />
        ).toJSON();

        expect(renderedComponent).toMatchSnapshot();
    });
});