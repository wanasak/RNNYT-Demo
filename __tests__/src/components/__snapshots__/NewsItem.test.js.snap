// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem component should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 3,
            "borderStyle": "solid",
          },
          Object {
            "marginBottom": 5,
          },
          Object {
            "borderColor": "#d31d65",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 100,
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://example.com/image.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "#34333699",
                "color": "#fff",
                "fontFamily": "HelveticaNeue-CondensedBold",
                "fontSize": 18,
              },
              Object {
                "padding": 5,
              },
            ]
          }
        >
          React Native Expands to New Platforms
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 11,
                },
                undefined,
              ]
            }
          >
            Sep 10th 2016
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 11,
                },
                undefined,
              ]
            }
          >
            JACOB FRIEDMANN
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#e5dbda",
              "fontFamily": "Helvetica Neue",
            },
            undefined,
          ]
        }
      >
        React Native, the framework for building mobile applications with web technologies, is expanding to new platforms.
      </Text>
    </View>
  </View>
</View>
`;
